npk/src/npk.rs

```diff
@@ -84,10 +84,32 @@ pub enum Error {
     },
 }
 
+/// Squashfs Options
+pub struct SquashfsOpts<'a> {
+    /// The compression algorithm used (default gzip)
+    pub comp: &'a str,
```

Comment:
Would be great to introduce a enum for the compression algorithm. This allows `clap` to hint if the argument is unknown.